test/create.ts

```diff
@@ -228,21 +228,21 @@ test('extend with custom handlers', withServer, async (t, server, got) => {
 			}
 		]
 	});
-	const headers = await instance('').json();
+	const headers = await instance('').json() as Headers;
 	t.is(headers.unicorn, 'rainbow');
 });
 
 test('extend with instances', t => {
 	const a = got.extend({prefixUrl: new URL('https://example.com/')});
 	const b = got.extend(a);
-	t.is(b.defaults.options.prefixUrl.toString(), 'https://example.com/');
+	t.is(b.defaults.options.prefixUrl!.toString(), 'https://example.com/');
 });
 
 test('extend with a chain', t => {
 	const a = got.extend({prefixUrl: 'https://example.com/'});
 	const b = got.extend(a, {headers: {foo: 'bar'}});
-	t.is(b.defaults.options.prefixUrl.toString(), 'https://example.com/');
-	t.is(b.defaults.options.headers.foo, 'bar');
+	t.is(b.defaults.options.prefixUrl!.toString(), 'https://example.com/');
+	t.is(b.defaults.options.headers!.foo, 'bar');
```

Comment:
`.headers` should always be defined. The default should just be an empty object.

Reply:
Fixed!